vm/src/stdlib/itertools.rs

```diff
@@ -537,6 +537,66 @@ impl PyItertoolsFilterFalse {
     }
 }
 
+#[pyclass]
+#[derive(Debug)]
+struct PyItertoolsAccumulate {
+    iterable: PyObjectRef,
+    binop: PyObjectRef,
+    acc_value: RefCell<PyObjectRef>,
```

Comment:
Maybe have this as an `Option` rather than using `None` as "not yet started iterating", because the binop function could return `None`.

Reply:
Right, this wasn't carefully thought out.